test/unit/math/rev/core/init_test.cpp

```diff
@@ -0,0 +1,62 @@
+#include <gtest/gtest.h>
+#include <stan/math/rev/core.hpp>
+
+#include <thread>
+
+TEST(init, thread_initialize) {
+  // the main thread must be initialized by the time this code is
+  // reached
+  EXPECT_TRUE(stan::math::ChainableStack::instance_ != nullptr);
+
+  stan::math::ChainableStack::AutodiffStackStorage& main_ad_stack
+      = stan::math::ChainableStack::instance();
```

Comment:
Should this line not just call our init() function?

Reply:
No, I don't think so. The test is execute after `init` has been called, since I am linking into the tests the `chainablestack_inst.cpp`. Thus, when the test starts, the stack is initialized in the main process only. Hence, the `instance_` is pointing somewhere (line 9 check) and calling `instance()` must give me the same reference as the result from `init` dereferenced.